test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua

```diff
@@ -0,0 +1,70 @@
+---------------------------------------------------------------------------------------------------
+-- https://github.com/smartdevicelink/sdl_core/issues/1398
+--
+-- Preconditions:
+-- 1) SDL and HMI are started
+-- 2) Mobiles №1 and №2 are connected to SDL
+--
+-- Tests:
+-- When SDL receives RegisterAppInterface RPC from mobile app, SDL must validate "appName" at first and validate the "appID" at second.
+-- In case the app registers with the same "appName" and different "appID" as the already registered one, SDL must return "resultCode: DUPLICATE_NAME, success: false" to such app.
+-- In case the app registers with the same "appName" and the same "appID" as the already registered one, SDL must return "resultCode: DUPLICATE_NAME, success: false" to such app.
+-- In case the app registers with the same "appName" as one of already registered "vrSynonyms" of other apps, SDL must return "resultCode: DUPLICATE_NAME, success: false" to such app. (that is, appName should not coinside with any of VrSynonims of already registered apps)
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/TheSameApp/commonTheSameApp')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Data ]]
+local devices = {
+  [1] = { host = "1.0.0.1", port = config.mobilePort }
+}
+
+local appParams = {
+  [1] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000001", vrSynonyms = {"VRSYNONYM1"} },
+  [2] = { appName = "EXISTING2", appID = "0002",  fullAppID = "00000002", vrSynonyms = {"VRSYNONYM2"} },
+  [3] = { appName = "EXISTING3", appID = "0003",  fullAppID = "00000003", vrSynonyms = {"VRSYNONYM3"} },
+  [4] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000004" },
```

Comment:
```suggestion
  [4] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000001" },
```

Reply:
This case specifies that short appID is the same, not the policyAppId. Such a change would result in APPLICATION_ALREADY_REGISTERED.

Reply:
`appID` is exclusively defined as a shortened version of `fullAppID` to account for systems which couldn't handle long app IDs. Back when this issue was introduced, there was no notion of `appID` and `fullAppID`, it used to just be `appID`. 

The case argues that duplicate names should be checked before the app ID, so I believe that it is saying that DUPLICATE_NAME should be returned if both are conflicting. If this is not the case now, then this issue is still technically present. That said, I think nitpicking over the result code in this case is unnecessary, so we can maybe argue that the issue is still invalid.

Reply:
Enlightening. Thank you for explaining.

In https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2403/commits/a6645251bef407988311f115aa40579aa6d5082e I have removed all mention of `fullAppID` for clarity and updated the expectation to be for `APPLICATION_REGISTERED_ALREADY` when both the name and ID of a new app match an existing one.

I consider the other issue to still be fixed as it describes this response as DISALLOWED, this other status is very descriptive and matches the error.